pkg/interceptors/bazel_flags.go

```diff
@@ -0,0 +1,88 @@
+/*
+Copyright © 2021 Aspect Build Systems Inc
+
+Not licensed for re-use.
+*/
+
+package interceptors
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"aspect.build/cli/pkg/bazel"
+	"github.com/spf13/cobra"
+)
+
+type Value interface {
+	String() string
+	Set(string) error
+	Type() string
+}
+
+type MultiString struct {
+	value *[]string
+}
+
+func (s *MultiString) Set(value string) error {
+	*s.value = append(*s.value, value)
+	return nil
+}
+
+func (s *MultiString) Type() string {
+	return "multiString"
+}
+
+func (s *MultiString) String() string {
+	return fmt.Sprintf("[ %s ]", strings.Join(*s.value, ", "))
+}
+
+func (s *MultiString) First() string {
+	return (*s.value)[0]
+}
+
+func BazelFlagInterceptor() Interceptor {
+	bzl := bazel.New()
+	return bazelflagInterceptor(bzl)
+}
+
+func bazelflagInterceptor(bzl bazel.Bazel) Interceptor {
+	return func(ctx context.Context, cmd *cobra.Command, args []string, next RunEContextFn) error {
+		workspaceRoot := ctx.Value(WorkspaceRootKey).(string)
+		bzl.SetWorkspaceRoot(workspaceRoot)
```

Comment:
Seems like currently we do this in every single `Command{RunE}`. Now we also do it within this interceptor?

What happens when `SetWorkspaceRoot` is set multiple times? Seems like the world would explode if the workspace root changes after the initial setup.

If every single `RunE` requires running `SetWorkspaceRoot`, and now this interceptor requires it... can the `WorkspaceRootInterceptor` not just set it once like it gets set here? That means any interceptor that requires `bzl.SetWorkspaceRoot()` being invoked must be run after the `WorkspaceRootInterceptor`... but that makes sense to me?